src/app/main.cpp

```diff
@@ -166,56 +174,35 @@ int main( int argc, char **argv )
 #else
-  ( void ) enableSentry;
+  Q_UNUSED( enableSentry );
 #endif
 
-  if ( !customLanguage.isEmpty() )
-    QgsApplication::setTranslation( customLanguage );
-
-  PlatformUtilities::instance()->initSystem();
-
-  QGuiApplication::setAttribute( Qt::AA_EnableHighDpiScaling );
   QtWebView::initialize();
 
-#if defined( Q_OS_ANDROID )
-  const QString projPath = PlatformUtilities::instance()->systemGenericDataLocation() + QStringLiteral( "/proj" );
+  if ( !customLanguage.isEmpty() )
+    QgsApplication::setTranslation( customLanguage );
 
-  const QDir rootPath = QStandardPaths::writableLocation( QStandardPaths::AppDataLocation );
-  rootPath.mkdir( QStringLiteral( "qgis_profile" ) );
-  const QString profilePath = QStandardPaths::writableLocation( QStandardPaths::AppDataLocation ) + QStringLiteral( "/qgis_profile" );
-  QgsApplication app( argc, argv, true, profilePath, QStringLiteral( "mobile" ) );
+  PlatformUtilities *platformUtils = PlatformUtilities::instance();
+  platformUtils->initSystem();
 
   QSettings settings;
+  // Let's make sure we have a writable path for the qgis_profile on every platform
+  const QString profilePath = platformUtils->systemLocalDataLocation( QStringLiteral( "/qgis_profile" ) );
+  QDir().mkdir( profilePath );
 
-  app.setThemeName( settings.value( "/Themes", "default" ).toString() );
-  app.setPrefixPath( "" QGIS_INSTALL_DIR, true );
-  app.setPluginPath( PlatformUtilities::instance()->systemGenericDataLocation() + QStringLiteral( "/plugins" ) );
-  app.setPkgDataPath( PlatformUtilities::instance()->systemGenericDataLocation() + QStringLiteral( "/qgis" ) );
-
-  app.createDatabase();
-#elif defined( Q_OS_IOS )
-  const QString projPath = PlatformUtilities::instance()->systemGenericDataLocation() + QStringLiteral( "/proj/data" );
-
-  const QDir rootPath = QStandardPaths::writableLocation( QStandardPaths::AppDataLocation );
-  rootPath.mkdir( QStringLiteral( "qgis_profile" ) );
-  const QString profilePath = QStandardPaths::writableLocation( QStandardPaths::AppDataLocation ) + QStringLiteral( "/qgis_profile" );
   QgsApplication app( argc, argv, true, profilePath, QStringLiteral( "mobile" ) );
+  app.setThemeName( settings.value( "/Themes", "default" ).toString() );
   app.setPkgDataPath( PlatformUtilities::instance()->systemGenericDataLocation() + QStringLiteral( "/qgis" ) );
   app.createDatabase();
-#else
-  QgsApplication app( argc, argv, true );
-  QSettings settings;
-  app.setThemeName( settings.value( "/Themes", "default" ).toString() );
 
 #ifdef RELATIVE_PREFIX_PATH
+  app.setPrefixPath( app.applicationDirPath() + "/..", true );
   qputenv( "GDAL_DATA", QDir::toNativeSeparators( app.applicationDirPath() + "/../share/gdal" ).toLocal8Bit() );
   const QString projPath( QDir::toNativeSeparators( app.applicationDirPath() + "/../share/proj" ) );
-  app.setPrefixPath( app.applicationDirPath() + "/..", true );
 #else
-  const QString projPath;
-  app.setPrefixPath( CMAKE_INSTALL_PREFIX, true );
```

Comment:
@m-kuhn , if I revert this change, I get QField to properly load QGIS providers et cie.

Reply:
Confirmed, let me think of a better solution here

Reply:
FindQGIS.cmake is updated, it will now set `QGIS_PREFIX_PATH` and assumes this is at `location_of_qgiscore_library/../..`. To be confirmed if this breaks other builds

Reply:
Nice. Will test within the hour.